### PR TITLE
Added priv filter methods

### PR DESCRIPTION
**Pull request by MadelineMurray**
Added priv filter methods

### CODE REVIEW

File: docs/Reference/API-Methods.md [hunk 1]
```diff
@@ -5001,6 +4999,129 @@ specified group of sender and recipients.
...
     }
     ```
+### priv_getFilterChanges
+
+Polls the specified filter for a private contract and returns an array of changes that have occurred 
+since the last poll.
+
+Privacy groups do not have blocks and private transactions cannot be pending so unlike 
+[`eth_getFilterChanges`](#eth_getfilterlogs), `priv_getFilterChanges` always returns an array 
+of log objects or an empty list.
```
Comment (lucassaldanha) [inline docs/Reference/API-Methods.md:5009]: Not sure if we need to mention it here, but it returns an error if the filter doesn't exist.

File: docs/Reference/API-Methods.md [hunk 2]
```diff
@@ -5001,6 +4999,129 @@ specified group of sender and recipients.
...
+
+Privacy groups do not have blocks and private transactions cannot be pending so unlike 
+[`eth_getFilterChanges`](#eth_getfilterlogs), `priv_getFilterChanges` always returns an array 
+of log objects or an empty list. 
+
+#### Parameters
+
+`data` - 32-byte [privacy Group ID](../Concepts/Privacy/Privacy-Groups.md).
+
+`data` - Filter ID hash
```
Comment (lucassaldanha) [inline docs/Reference/API-Methods.md:5015]: Drop the hash. It is just an ID :)

File: docs/Reference/API-Methods.md [hunk 3]
```diff
@@ -5001,6 +4999,129 @@ specified group of sender and recipients.
...
+!!! note
+
+    `priv_getFilterLogs` is only used for filters created with [`priv_newFilter`](#priv_newfilter). 
+    To specify a filter object and get logs without creating a filter, use `priv_getLogs`.
+
+#### Parameters
+
+`data` - 32-byte [privacy Group ID](../Concepts/Privacy/Privacy-Groups.md).
+
+`data` - Filter ID hash.
```
Comment (lucassaldanha) [inline docs/Reference/API-Methods.md:5073]: s.a.

File: docs/Reference/API-Methods.md [hunk 4]
```diff
@@ -5277,6 +5398,86 @@ or `null` if no receipt found.
     }
     ```
 
+### priv_newFilter
+
+Creates a [log filter](../Concepts/Events-and-Logs.md) for a private contract. To poll for logs associated with the
+created filter, use [priv_getFilterChanges](#priv_getfilterchanges).
```
Comment (lucassaldanha) [inline docs/Reference/API-Methods.md:5404]: You can use `priv_getFilterChanges` or `priv_getFilterLogs` to retrieve logs.

File: docs/Reference/API-Methods.md [hunk 5]
```diff
@@ -5277,6 +5398,86 @@ or `null` if no receipt found.
...
+`Object` - [Filter options object](API-Objects.md#filter-options-object).
+
+!!! note
+
+    `fromBlock` and `toBlock` in the filter options object default to `latest`. To obtain logs
+    using [`priv_getFilterLogs`](#priv_getfilterlogs), set `fromBlock` and `toBlock` appropriately.
+
+#### Returns
+
+`data` - Filter ID hash.
```
Comment (lucassaldanha) [inline docs/Reference/API-Methods.md:5422]: s.a.

File: docs/Reference/API-Methods.md [hunk 6]
```diff
@@ -5277,6 +5398,86 @@ or `null` if no receipt found.
...
+minutes.
+
+For private contracts, `priv_uninstallFilter` is the same as [`eth_uninstallFilter`](#eth_uninstallfilter)
+for public contracts. 
+
+#### Parameters
+
+`data` - 32-byte [privacy Group ID](../Concepts/Privacy/Privacy-Groups.md).
+
+`data` - Filter ID hash.
```
Comment (lucassaldanha) [inline docs/Reference/API-Methods.md:5457]: s.a.

File: mkdocs.yml [hunk 7]
```diff
@@ -212,8 +212,8 @@ markdown_extensions:
     # pymdownx.superfences is deprecated.
     # It must be replaced with pymdownx.tabbed but it requires new CSS
     # TODO with material 5 update
-    - pymdownx.superfences:
-        legacy_tab_classes: true
+    - pymdownx.superfences
```
Comment (lucassaldanha) [inline mkdocs.yml:215]: I'm not sure what this change means.

File: docs/Reference/API-Methods.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5009]: We don't explicitly state this for other methods.  If we have sensible error messages, it should be clear to the user what's happened without needed to duplicate this in the doc.

File: docs/Reference/API-Methods.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5015]: Done throughout.  This was copied over from the eth filter methods so have fixed there as well.

File: docs/Reference/API-Methods.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5073]: Fixed

File: docs/Reference/API-Methods.md (hunk 4 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5404]: Added (and for eth_newFilter)

File: docs/Reference/API-Methods.md (hunk 5 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5422]: Fixed

File: docs/Reference/API-Methods.md (hunk 6 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5457]: Fixed

File: mkdocs.yml (hunk 7 above)
Comment (MadelineMurray) [reply mkdocs.yml:215]: Publishing stuff

File: docs/Reference/API-Methods.md [hunk 8]
```diff
@@ -5342,6 +5465,87 @@ or `null` if no receipt found.
...
+#### Parameters
+
+`data` - 32-byte [privacy Group ID](../Concepts/Privacy/Privacy-Groups.md).
+
+`Object` - [Filter options object](API-Objects.md#filter-options-object).
+
+!!! note
+
+    `fromBlock` and `toBlock` in the filter options object default to `latest`. To obtain logs
+    using [`priv_getFilterLogs`](#priv_getfilterlogs), set `fromBlock` and `toBlock` appropriately.
```
Comment (bgravenorst) [inline docs/Reference/API-Methods.md:5486]: Could probably drop "appropriately".

File: docs/Reference/API-Methods.md [hunk 9]
```diff
@@ -5002,6 +5001,130 @@ specified group of sender and recipients.
...
+contract.
+
+For private contracts, `priv_getFilterLogs` is the same as [`eth_getFilterLogs`](#eth_getfilterlogs)
+for public contracts except there is no [automatic log bloom caching](CLI/CLI-Syntax.md#auto-log-bloom-caching-enabled)
+for private contracts.
+
+!!! note
+
+    `priv_getFilterLogs` is only used for filters created with [`priv_newFilter`](#priv_newfilter).
+    To specify a filter object and get logs without creating a filter, use `priv_getLogs`.
```
Comment (bgravenorst) [inline docs/Reference/API-Methods.md:5069]: Link to `priv_getLogs`?

File: mkdocs.yml [hunk 10]
```diff
@@ -213,8 +213,8 @@ markdown_extensions:
...
+        #legacy_tab_classes: true
```
Comment (bgravenorst) [inline mkdocs.yml:217]: Does this not break the fencing?

File: docs/Reference/API-Methods.md (hunk 8 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5486]: Dropped sentence

File: docs/Reference/API-Methods.md (hunk 9 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md:5069]: Added

File: mkdocs.yml (hunk 10 above)
Comment (MadelineMurray) [reply mkdocs.yml:217]: Commented out per @NicolasMassart to enable building.  Have double checked local publishing and it's OK.

File: mkdocs.yml (hunk 7 above)
Comment (NicolasMassart) [reply mkdocs.yml:216]: It broke the tabs, so it was then reversed in another PR #368. This should not have been changed. If you have any doubt on things like this, ask.

File: mkdocs.yml (hunk 10 above)
Comment (NicolasMassart) [reply mkdocs.yml:217]: Reverted in #368
I was only talking about disabling for local build not for final readthedocs build. Sorry for the misunderstanding.

**Pull request merged**